test/system/272-system-connection.bats

```diff
@@ -180,4 +180,73 @@ $c2[ ]\+tcp://localhost:54321[ ]\+true" \
     run_podman system connection rm mysshconn
 }
 
+# Check the priority of --connection flag vs CONTAINER_HOST env variable
+@test "podman-remote: non-default connection" {
```

Comment:
@edsantiago PTAL at the test, I am not sure if you have a better idea to structure this: the full matrix is quite complex.